src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidBrokerHelper.cs

```diff
@@ -208,31 +120,30 @@ public void CheckForBrokerAccountInfoInAccountManager(BrokerRequest brokerReques
             string homeAccountId = brokerRequest.HomeAccountId;
             string localAccountId = brokerRequest.LocalAccountId;
 
-            if (!string.IsNullOrEmpty(accounts))
+            if (!string.IsNullOrEmpty(accountData))
             {
-                dynamic authResult = JArray.Parse(accounts);
+                dynamic authResult = JArray.Parse(accountData);
 
                 foreach (JObject account in authResult)
                 {
-                    var accountData = account[BrokerResponseConst.Account];
-
-                    var accountDataHomeAccountID = accountData[BrokerResponseConst.HomeAccountId]?.ToString();
-                    var accountDataLocalAccountID = accountData[BrokerResponseConst.LocalAccountId]?.ToString();
+                    var accountInfo = account[BrokerResponseConst.Account];
+                    var accountInfoHomeAccountID = accountInfo[BrokerResponseConst.HomeAccountId]?.ToString();
+                    var accountInfoLocalAccountID = accountInfo[BrokerResponseConst.LocalAccountId]?.ToString();
 
-                    if (string.Equals(accountData[BrokerResponseConst.UserName].ToString(), username, StringComparison.OrdinalIgnoreCase))
+                    if (string.Equals(accountInfo[BrokerResponseConst.UserName].ToString(), username, StringComparison.OrdinalIgnoreCase))
                     {
                         // TODO: broker request should be immutable!
-                        brokerRequest.HomeAccountId = accountDataHomeAccountID;
-                        brokerRequest.LocalAccountId = accountDataLocalAccountID;
+                        brokerRequest.HomeAccountId = accountInfoHomeAccountID;
+                        brokerRequest.LocalAccountId = accountInfoLocalAccountID;
```

Comment:
why not use the variables created before and is it safe to override the value here?

Reply:
Yes it is safe to overwrite. for the silent request, these values start out as null because we need to fetch them from the broker first. Then we populate